pkg/collector/corechecks/snmp/fetch_column.go

```diff
@@ -0,0 +1,117 @@
+package snmp
+
+import (
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/gosnmp/gosnmp"
+	"sort"
+)
+
+func fetchColumnOidsWithBatching(session sessionAPI, oids map[string]string, oidBatchSize int) (columnResultValuesType, error) {
+	retValues := make(columnResultValuesType, len(oids))
+
+	columnOids := getOidsMapKeys(oids)
+	sort.Strings(columnOids) // sorting columnOids to make them deterministic for testing purpose
```

Comment:
Alternatively, if these aren't expected to change much, can we cache them and refresh occasionally? I feel like we already discussed this in our zoom review, so feel free to ignore if this is a stupid question.

Reply:
That's in the hot path and I think you are right, there are some possible improvement here.

- It's a good idea to use caching here and store it on `snmpConfig.oidConfig` attached to the check instance. Instead of storing list of oids in `snmpConfig.oidConfig` we can store already batched OIDs.
- To make the batches predictable, we can store in a list instead of a map. That will get rid of the `sort.Strings`
- The refresh will happen when `snmpConfig.refreshWithProfile` is called (only once for 99% cases)

Do you mind if I move this optimisation as a separate PR ? (will create a specific issue ticket for that.

Reply:
Isn't there a `SortedMap` data structure in golang that let sorted storage and constant time element access by key? I think using a list would decrease performances when looking for a specific oid - I think it's done at enrichment time to get the list of tags, isn't it?

Reply:
> Isn't there a SortedMap data structure in golang that let sorted storage and constant time element access by key? 

Found this https://golang.org/pkg/internal/fmtsort/ 
but it's only for internal use. Possibly because of a higher cost of the data structure.

> I think using a list would decrease performances when looking for a specific oid - I think it's done at enrichment time to get the list of tags, isn't it?

In this case, we don't need to make lookups, we mainly need to loop/batch the elements. I probably could have used a list in the first place.

Reply:
Created II-199 to track this change.